suzieq/poller/writers/output_worker.py

```diff
@@ -15,19 +16,21 @@ class OutputWorker(SqPlugin):
     implementing the logic to persist the output of the polling
     """
     def __init__(self, **kwargs):
-        self.type = kwargs.get("type", None)
+        self.type = kwargs.get('type', None)
         self.logger = logging.getLogger(__name__)
 
-        output_dir = kwargs.get("output_dir", None)
+        output_dir = kwargs.get('output_dir', None)
         if output_dir:
             self.root_output_dir = output_dir
-            if not os.path.isdir(output_dir):
+            if not os.path.exists(output_dir):
                 os.makedirs(output_dir)
+            elif not os.path.isdir(output_dir):
+                raise SqPollerConfError(f'Output directory {output_dir}'
+                                        'is not a directory')
         else:
             # TBD: The right error to raise here since this is a required
             # keyword
-            self.logger.error("Need mandatory keyword arg: output_dir")
-            raise ValueError
+            raise SqPollerConfError('Need mandatory keyword arg: output_dir')
```

Comment:
Is there a reason why you removed the log?
I think the comment right above should be removed

Reply:
Log is generated at the exception catch level